Add information about dynamically loading modules in JavaScript in the progressive loading article (#19952)
mdn/content: files/en-us/web/progressive_web_apps/loading/index.md

<script src="app.js" defer></script>
```

They will be downloaded and executed _after_ the document itself has been parsed, so it won't block rendering the HTML structure.

> Another technique is to load JavaScript modules, using [import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), only when needed.

Reviewer: ```suggestion
Another technique is to load JavaScript modules using [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) only when needed.
```

> Another technique is to load JavaScript modules using [dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) only when needed.

Reviewer: ```suggestion
Another technique is to load JavaScript modules using [dynamic import](/en-US/docs/Web/JavaScript/Reference/Operators/import) only when needed.
```

> For example, if a website has a search button we can load the JavaScript for the search function after the user clicks on the search button:

Reviewer: ```suggestion
For example, if a website has a search button, we can load the JavaScript for the search function after the user clicks on the search button:
```

```js

> document.getElementById("open-search").addEventListener("click", async () => {
>    let searchModule = await import ("/modules/search.js");
>    searchModule.loadAutoComplete();
> });

Reviewer: ```suggestion
document.getElementById("open-search").addEventListener("click", async () => {
  const searchModule = await import("/modules/search.js");
  searchModule.loadAutoComplete();
});
```

```

> Once the user clicks on the button the async arrow function is called. The function waits till the module is loaded then calls the `loadAutoComplete` function so the browser only loads the code when needed. 

Reviewer: ```suggestion
Once the user clicks on the button, the async click handler is called. The function waits till the module is loaded, then calls the `loadAutoComplete()` function exported from that module. The `search.js` module is only downloaded, parsed, and executed when needed.
```
